File: string_array/Jump_Game_II.java
class Solution {
    public int jump(int[] nums) {
        int currentIndex = 0;
        int nextIndex = 0;
        int jumps = 0;
        for (int i = 0; i < nums.length -1; ++i) {
            nextIndex = Math.max(nextIndex, i + nums[i]);
            if (nextIndex >= nums.length - 1) {
                return ++jumps;
            }
            if (i == currentIndex) {
                ++jumps;
                currentIndex = nextIndex;
            }
        }
        return jumps;
    }
}

File: string_array/Jump_Game_II.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        nextIndex = 0
        currentIndex = 0
        jumps = 0
        for i in range(0, len(nums) - 1):
            nextIndex = max(nextIndex, i + nums[i])
            if (nextIndex >= len(nums) - 1):
                jumps += 1
                return jumps
            if (i == currentIndex):
                jumps += 1
                currentIndex = nextIndex
        return jumps